virtual_ship/virtual_ship_configuration.py

```diff
@@ -1,173 +1,156 @@
+                    ]
+                )
+                for deploy in self.drifter_deploy_locations
+            ]
         ):
             raise ValueError(
-                "Specify negative depth. Max depth for argo is -5727 m due to data availability"
+                "Drifter deploy locations are not exactly on route coordinates."
             )
-        if type(self.argo_characteristics["vertical_speed"]) is not float:
-            raise ValueError("Specify vertical speed for argo with decimals in m/s")
-        if self.argo_characteristics["vertical_speed"] > 0:
-            self.argo_characteristics["vertical_speed"] = -self.argo_characteristics[
-                "vertical_speed"
+
+        if not all(
+            [self._is_valid_location(coord) for coord in self.ctd_deploy_locations]
+        ):
+            raise ValueError("CTD deploy locations are not valid coordinates.")
+
+        if not all(
+            [
+                any(
+                    [
+                        np.isclose(deploy.lat, coord.lat)
+                        and np.isclose(deploy.lon, coord.lon)
+                        for coord in self.route_coordinates
+                    ]
+                )
+                for deploy in self.ctd_deploy_locations
             ]
-        if (
-            abs(self.argo_characteristics["vertical_speed"]) < 0.06
-            or abs(self.argo_characteristics["vertical_speed"]) > 0.12
         ):
             raise ValueError(
-                "Specify a realistic speed for argo, i.e. between -0.06 and -0.12 m/s"
+                "CTD deploy locations are not exactly on route coordinates."
             )
-        if self.argo_characteristics["cycle_days"] < 0:
-            raise ValueError("Specify a postitive number of cycle days for argo")
-        if self.argo_characteristics["drift_days"] < 0:
-            raise ValueError("Specify a postitive number of drift days for argo")
+
+        if self.argo_float_config.max_depth > 0:
+            raise ValueError("Argo float max depth must be negative or zero.")
+
+        if self.argo_float_config.drift_depth > 0:
+            raise ValueError("Argo float drift depth must be negative or zero.")
+
+        if self.argo_float_config.vertical_speed >= 0:
+            raise ValueError("Argo float vertical speed must be negative.")
+
+        if self.argo_float_config.cycle_days <= 0:
+            raise ValueError("Argo float cycle days must be large than one.")
+
+        if self.argo_float_config.drift_days <= 0:
+            raise ValueError("Argo drift cycle days must be large than one.")
+
+        if self.adcp_config.max_depth > 0:
+            raise ValueError("ADCP max depth must be negative.")
+
+    @staticmethod
+    def _is_valid_location(location: Location) -> bool:
+        return (
+            location.lat >= -90
+            and location.lat <= 90
+            and location.lon >= -180
+            and location.lon <= 180
```

Comment:
Hmm, this may be too strict. Many people (including myself) are used to longitudes between 0 and 360 degrees